flang/lib/Lower/Bridge.cpp

```diff
@@ -2162,6 +2226,10 @@ class FirConverter : public Fortran::lower::AbstractConverter {
 
     // This call may generate a branch in some contexts.
     genFIR(endDoEval, unstructuredContext);
+
+    // Add attribute(s) on operations in fir::DoLoopOp if necessary
```

Comment:
Why is there special handling for `fir.do_loop` but not for `fir.if` and other operations with regions?

Reply:
For the moment, I've focused on do_loop, assignment and callstmt. And in `doc/Directives.md` I've added a message to say that these directives are partially supported. But if it's really necessary I can add support for the other operations.